go-controller/pkg/ovn/pods.go

```diff
@@ -371,6 +414,21 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 	}
 	lspExist = err != libovsdbclient.ErrNotFound
 
+	// Sanity check. If port exists, it should be in the logical switch obtained from the pod spec.
+	if lspExist {
+		portFound := false
+		for _, currPortUUID := range ls.Ports {
+			if currPortUUID == existingLSP.UUID {
+				portFound = true
+				break
+			}
+		}
+		if !portFound {
```

Comment:
@trozet we could actually return the error here, since this call is part of the retry_obj... wdyt?

Reply:
yeah I think that is a good idea. Eventually maybe we need to add a retry delete entry as well, but we probably need to give that more thought before we implement it.

Reply:
@flavio-fernandes since looks like you are on PTO today I'll just change this to return an error msg for you